src/vs/workbench/contrib/scm/browser/scm.contribution.ts

```diff
@@ -355,6 +355,16 @@ Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Configuration).regis
 			maximum: 1000,
 			default: 50
 		},
+		'scm.graph.labels': {
+			type: 'string',
+			enum: ['all', 'filter'],
+			enumDescriptions: [
+				localize('scm.graph.labels.all', "Show all labels in the Source Control Graph view."),
+				localize('scm.graph.labels.filter', "Show only the labels of the history item groups used as a filter in the Source Control Graph view.")
+			],
+			description: localize('scm.graph.labels', "Controls the labels of the history item groups shown in the Source Control Graph view."),
```

Comment:
"Labels" are the visual indicators of the ref names on the right hand side. The UX is still very much in flux so if you think we have a better name for these I am happy to update the setting. 
<img width="415" alt="image" src="https://github.com/user-attachments/assets/8ab87f0e-b5d4-4306-b771-17b253857711">

Reply:
Oh ok, I was thinking it was the commit message since that's what label/title/name/description normally refer to in lists. A better name might be "badge"? The more important change imo is to clarify in the description:

> Controls the labels of the history item groups shown in the Source Control Graph view. **These are the icons on the right side of the graph that indicate things like origin, branches, etc.**

Reply:
I like badge, especially that we are considering tweaking the design to be more pill-like...